pkg/prometheus/promcfg.go

```diff
@@ -272,6 +294,248 @@ func (cg *configGenerator) generateConfig(
+				{Key: "regex", Value: strings.Join(exp.Values, "|")},
+			})
+		case metav1.LabelSelectorOpExists:
+			relabelings = append(relabelings, yaml.MapSlice{
+				{Key: "action", Value: "keep"},
+				{Key: "source_labels", Value: []string{"__meta_kubernetes_pod_label_" + sanitizeLabelName(exp.Key)}},
+				{Key: "regex", Value: ".+"},
+			})
+		case metav1.LabelSelectorOpDoesNotExist:
+			relabelings = append(relabelings, yaml.MapSlice{
+				{Key: "action", Value: "drop"},
+				{Key: "source_labels", Value: []string{"__meta_kubernetes_pod_label_" + sanitizeLabelName(exp.Key)}},
+				{Key: "regex", Value: ".+"},
+			})
+		}
+	}
+
+	if version.Major == 1 && version.Minor < 7 {
+		// Filter targets based on the namespace selection configuration.
+		// By default we only discover services within the namespace of the
+		// PodMonitor.
+		// Selections allow extending this to all namespaces or to a subset
+		// of them specified by label or name matching.
+		//
+		// Label selections are not supported yet as they require either supported
+		// in the upstream SD integration or require out-of-band implementation
+		// in the operator with configuration reload.
+		//
+		// There's no explicit nil for the selector, we decide for the default
+		// case if it's all zero values.
+		nsel := m.Spec.NamespaceSelector
+
+		if !nsel.Any && len(nsel.MatchNames) == 0 {
+			relabelings = append(relabelings, yaml.MapSlice{
+				{Key: "action", Value: "keep"},
+				{Key: "source_labels", Value: []string{"__meta_kubernetes_namespace"}},
+				{Key: "regex", Value: m.Namespace},
+			})
+		} else if len(nsel.MatchNames) > 0 {
+			relabelings = append(relabelings, yaml.MapSlice{
+				{Key: "action", Value: "keep"},
+				{Key: "source_labels", Value: []string{"__meta_kubernetes_namespace"}},
+				{Key: "regex", Value: strings.Join(nsel.MatchNames, "|")},
+			})
+		}
+	}
+
+	// Filter targets based on correct port for the endpoint.
+	if ep.Port != "" {
+		relabelings = append(relabelings, yaml.MapSlice{
+			{Key: "action", Value: "keep"},
+			{Key: "source_labels", Value: []string{"__meta_kubernetes_pod_container_port_name"}},
+			{Key: "regex", Value: ep.Port},
+		})
+	} else if ep.TargetPort != nil {
+		if ep.TargetPort.StrVal != "" {
+			relabelings = append(relabelings, yaml.MapSlice{
+				{Key: "action", Value: "keep"},
+				{Key: "source_labels", Value: []string{"__meta_kubernetes_pod_container_port_name"}},
+				{Key: "regex", Value: ep.TargetPort.String()},
+			})
+		} else if ep.TargetPort.IntVal != 0 {
+			relabelings = append(relabelings, yaml.MapSlice{
+				{Key: "action", Value: "keep"},
+				{Key: "source_labels", Value: []string{"__meta_kubernetes_pod_container_port_number"}},
+				{Key: "regex", Value: ep.TargetPort.String()},
+			})
+		}
+	}
+
+	// Relabel namespace and pod and service labels into proper labels.
+	relabelings = append(relabelings, []yaml.MapSlice{
+		{
+			{Key: "source_labels", Value: []string{"__meta_kubernetes_pod_node_name"}},
```

Comment:
this is not something we currently do or encourage with ServiceMonitors, this should be done at query time against kube-state-metrics

Reply:
makes sense. dropped this relabeling